src/intern.ts

```diff
@@ -384,8 +384,33 @@ const intern = {
     const isUpdate = rowExist ? true : false
 
     return isUpdate
-  }
+  },
+
+  buildCtx(seneca: any, msg: any, meta: any) {
+    let ctx: any = {}
+    let transaction = seneca.entity.state().transaction
+
+    if(transaction && !transaction.finish && false !== msg.transaction$) {
+      transaction.trace.push({
+        when: Date.now(),
+        msg,
+        meta,
+      })
+      ctx.transaction = transaction
+      ctx.client = transaction.client
+    }
+    
+    return ctx
+  },
+  async getKnexClient(knex: any, seneca: any, msg: any) {
```

Comment:
be careful to maintain consistent code formatting
there should be a blank line here

Reply:
Done

Reply:
done